src/JustNotSorry.js

```diff
@@ -1,51 +1,72 @@
+          var inputEvent = new Event('input', {
+              bubbles: true,
+              cancelable: true,
+          });
+          targetDiv.dispatchEvent(inputEvent);
+        }
+      }
+    });
   }
+});
+
+var addObserver = function() {
+  warningChecker.addWarnings(this.parentNode);
+  observer.observe(this, {characterData: true, subtree: true, childList: true,  attributes: true});
 }
 
-var warningChecker = new WarningChecker(WARNINGS);
+var removeObserver = function() {
+  warningChecker.removeWarnings(this.parentNode);
+  observer.disconnect();
+}
 
-var addTextEventListener = function(mutation) {
-  ['focus', 'input'].forEach(function(action) {
-    document.querySelector('div[contentEditable=true]').addEventListener(action, checkForWarnings(warningChecker, mutation))
-  });
+var checkForWarnings = function() {
+  warningChecker.removeWarnings(this.parentNode);
+  warningChecker.addWarnings(this.parentNode);
 }
 
-var removeTextEventListener = function() {
-  ['focus', 'input'].forEach(function(action) {
-    document.querySelector('div[contentEditable=true]').removeEventListener(action, checkForWarnings(warningChecker, action))
-  });
+var applyEventListeners = function(id) {
+  var targetDiv = document.getElementById(id);
+  targetDiv.addEventListener('focus', addObserver);
+  targetDiv.addEventListener('input', checkForWarnings);
+  targetDiv.addEventListener('blur', removeObserver);
 }
 
-var observer = new MutationObserver(function(mutation) {
-  if (document.querySelector('div[contentEditable=true]')) {
-    addTextEventListener(mutation);
-    removeTextEventListener();
+var documentObserver = new MutationObserver(function(mutations) {
+  var divCount = getEditableDivs().length;
+  if (divCount != editableDivCount) {
+    editableDivCount = divCount;
+    var id;
+    if (mutations[0]) {
+      mutations.forEach(function(mutation){
+        if (mutation.type == 'childList' && mutation.target.hasAttribute('contentEditable')) {
+          id = mutation.target.id;
+          if (id) {
+            applyEventListeners(id);
+          }
+        }
+      });
+    }
   }
 });
 
-function checkForWarnings(warningChecker, mutation) {
-  var target
-  target = document.querySelector('div[contentEditable=true]');
-
-  document.querySelectorAll('div[contentEditable=true]').forEach((field) => {
-    var active = document.activeElement;
-    [field, active].reduce((a, b) => {
-      if (a != b) {
-        target = active;
-      }
-      if (target === null) {
-        target = field;
-      }
-    });
-  });
-
-  warningChecker.removeWarnings(target.parentNode);
-  warningChecker.addWarnings(target.parentNode);
-  removeWarningsOnBlur(target.parentNode);
+function getEditableDivs() {
+  return document.querySelectorAll('div[contentEditable=true]');
 }
 
-observer.observe(document, {characterData: true, subtree: true})
+documentObserver.observe(document, {characterData: true, subtree: true, childList: true});
```

Comment:
It seems like the documentObserver wouldn't need to be monitoring the characterData

Reply:
Good catch, it indeed wasn't necessary.